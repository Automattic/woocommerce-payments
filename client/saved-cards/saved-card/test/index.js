/** @format */

/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { SavedCard } from '../';

const mockCard = {
	tokenId: 1,
	paymentMethodId: 'pm_mock_1',
	isDefault: true,
	brand: 'visa',
	last4: '4242',
	expiryMonth: '04',
	expiryYear: '2024',
};

describe( 'Saved Card', () => {
	beforeEach( () => {
		jest.clearAllMocks();
	} );

	test( 'renders default card correctly', () => {
		const { container } = render( <SavedCard { ...mockCard } /> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders non-default card correctly', () => {
		const { container } = render(
			<SavedCard { ...mockCard } { ...{ isDefault: false } } />
		);
		expect( container ).toMatchSnapshot();
	} );
} );
