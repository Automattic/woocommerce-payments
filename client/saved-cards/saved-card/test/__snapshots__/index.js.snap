// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saved Card renders default card correctly 1`] = `
<div>
  <div
    class="wcpay-saved-cards__card undefined"
  >
    <span
      class="payment-method__brand payment-method__brand--visa"
    />
    <span
      class="wcpay-saved-cards__card-info"
    >
      <span
        class="wcpay-saved-cards__card-info--card"
      >
        Visa •••• 4242
      </span>
      <span
        class="wcpay-saved-cards__card-info--expiry"
      >
        Expires Apr 2024
      </span>
    </span>
    <span
      class="wcpay-saved-cards__card-default"
    >
      <span
        class="chip chip-primary"
      >
        Default
      </span>
    </span>
    <span
      class="wcpay-saved-cards__card-id"
    >
      <span
        class="label"
      >
        Card ID:
      </span>
       
      pm_mock_1
    </span>
  </div>
</div>
`;

exports[`Saved Card renders non-default card correctly 1`] = `
<div>
  <div
    class="wcpay-saved-cards__card undefined"
  >
    <span
      class="payment-method__brand payment-method__brand--visa"
    />
    <span
      class="wcpay-saved-cards__card-info"
    >
      <span
        class="wcpay-saved-cards__card-info--card"
      >
        Visa •••• 4242
      </span>
      <span
        class="wcpay-saved-cards__card-info--expiry"
      >
        Expires Apr 2024
      </span>
    </span>
    <span
      class="wcpay-saved-cards__card-default"
    />
    <span
      class="wcpay-saved-cards__card-id"
    >
      <span
        class="label"
      >
        Card ID:
      </span>
       
      pm_mock_1
    </span>
  </div>
</div>
`;
